plutus-use-cases/src/Plutus/Contracts/Crowdfunding.hs

```diff
@@ -190,10 +188,10 @@ crowdfunding :: Campaign -> Contract () CrowdfundingSchema ContractError ()
 crowdfunding c = contribute c `select` scheduleCollection c
 
 -- | A sample campaign
-theCampaign :: Campaign
-theCampaign = Campaign
-    { campaignDeadline = TimeSlot.slotToEndPOSIXTime def 20
-    , campaignCollectionDeadline = TimeSlot.slotToEndPOSIXTime def 30
+theCampaign :: POSIXTime -> Campaign
+theCampaign startTime = Campaign
+    { campaignDeadline = startTime + 20999
```

Comment:
What's `20999`?

Reply:
Also, since this is just an example value of `Campaign`, we could just choose a start time here, instead of having it as a parameter. This would simplify the rest of the code quite a bit.

Reply:
`20999` ms is the moment just before `21s`. So at `21s`, the crowdfunding campaign is over. So `20999ms` represents the last moment were the campaign is still on.

Reply:
Should I change it so that we should use round numbers?

Reply:
Well, or this should be the time where it ends rather than the last time where it's active, and we should have a `<` rather than a `<=` in the on-chain code!